app/src/main/java/br/com/mob1st/bet/core/utils/objects/Node.kt

```diff
@@ -1,9 +1,15 @@
 package br.com.mob1st.bet.core.utils.objects
 
+
+import androidx.annotation.Keep
+import kotlinx.serialization.Serializable
+
 /**
  * A tree node that indicates the [current] value in the tree and the next possible [paths]
  */
-data class Node<T>(
+@Serializable
+@Keep
+data class Node<out T>(
```

Comment:
O que é esse out?

Reply:
acho que fica mais facil explicar por código
```kotlin
// normal usage
interface Sample<T> {
    // it will work
    fun into(value: T)

    // it will work
    fun get(): T
}

// apenas "in" permitido
interface Sample<in T> {
    // it will work
    fun into(value: T)

    // it will NOT work
    fun get(): T
}

// apenas "out" permitido
interface Sample<out T> {
    // it will NOT work
    fun into(value: T)

    // it will work
    fun get(): T
}
```

é bem raro precisar usar isso. eu usei nesse exemplo pq em um caso específico o android studio tava pedindo.
se quiser mais detalhes pode ler [aqui](https://kotlinlang.org/docs/generics.html)